src/EditorFeatures/TestUtilities/Classification/AbstractClassifierTests.cs

```diff
@@ -226,21 +263,11 @@ public abstract class AbstractClassifierTests
 
         protected static async Task<ImmutableArray<ClassifiedSpan>> GetSemanticClassificationsAsync(Document document, TextSpan span)
         {
-            var tree = await document.GetSyntaxTreeAsync();
-
-            var service = document.GetLanguageService<ISyntaxClassificationService>();
-            var classifiers = service.GetDefaultSyntaxClassifiers();
-            var extensionManager = document.Project.Solution.Workspace.Services.GetService<IExtensionManager>();
+            var service = document.GetRequiredLanguageService<IClassificationService>();
 
-            var results = ArrayBuilder<ClassifiedSpan>.GetInstance();
-
-            await service.AddSemanticClassificationsAsync(document, span,
-                extensionManager.CreateNodeExtensionGetter(classifiers, c => c.SyntaxNodeTypes),
-                extensionManager.CreateTokenExtensionGetter(classifiers, c => c.SyntaxTokenKinds),
-                results,
-                CancellationToken.None);
-
-            return results.ToImmutableAndFree();
+            var result = new List<ClassifiedSpan>();
+            await service.AddSemanticClassificationsAsync(document, span, result, CancellationToken.None);
+            return result.ToImmutableArray();
```

Comment:
test code wasn't actually going through the interface (it was hardcoding what the impl did). So this wasn't actually calling through OOP until i fixed this up.